src/test/java/test/listeners/ListenerTest.java

```diff
@@ -124,6 +124,15 @@ public void methodInterceptorShouldBeRunOnce() {
     tng.run();
     Assert.assertEquals(interceptor.getCount(), 1);
   }
+  
+  @Test(description = "GITHUB-1863:IMethodInterceptor will be invoked twice when listener implements both ITestListener and IMethodInterceptor via eclipse execution way")
+  public void methodInterceptorShouldBeRunOnceWhenCustomisedListenerImplementsITestListenerAndIMethodInterceptor() {
+    TestNG tng = create(LSampleTest.class);
+    InterceptorInvokeTwiceSimulateListener interceptor = new InterceptorInvokeTwiceSimulateListener();
+    tng.addListener((ITestNGListener) interceptor);
```

Comment:
The ITestNGListener cast should be useless now. Could you confirm?

Reply:
Yes, case is unnecessary, removed.